android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java

```diff
@@ -24,14 +26,12 @@
     private static final String TAG = "PeggiMeter.MoodAdapter";
 
     private final DatabaseReference moods;
-    private final DataController dataController;
     private final List<LogEntry> logs;
-    private final List<MoodListener> listeners;
+    private final Set<MoodListener> listeners;
 
     MoodAdapter(DataController dataController) {
-        this.dataController = dataController;
         this.logs = new ArrayList<>();
-        this.listeners = new ArrayList<>();
+        this.listeners = new HashSet<>();
```

Comment:
Do you have any specific need for a set here? If so please make sure we override hashCode and equals methods on implementation.

Reply:
now I do

Reply:
Please do the same for the second adapter. Thanks!